Gems/ROS2/Code/Source/RobotImporter/Utils/SourceAssetsStorage.cpp

```diff
@@ -541,13 +569,46 @@ namespace ROS2::Utils
         scene->GetManifest().SaveToFile(assetInfoFile.Native());
         AZ_Printf("CreateSceneManifest", "Saving scene manifest to %s\n", assetInfoFile.c_str());
 
+        assetsFilepaths = GetMeshAssets(scene);
+        AZ_Printf("CreateSceneManifest", "Found %d assets for %s scene\n", assetsFilepaths.size(), assetInfoFile.c_str());
+
         return true;
     }
 
-    bool CreateSceneManifest(const AZ::IO::Path& sourceAssetPath, bool collider, bool visual)
+    bool CreateSceneManifest(
+        const AZ::IO::Path& sourceAssetPath, const bool collider, const bool visual, AZStd::unordered_set<AZ::IO::Path>& assetsFilepaths)
+    {
+        return CreateSceneManifest(sourceAssetPath, sourceAssetPath.Native() + ".assetinfo", collider, visual, assetsFilepaths);
+    }
+
+    AZStd::unordered_set<AZ::IO::Path> GetMeshAssets(const AZStd::shared_ptr<AZ::SceneAPI::Containers::Scene>& scene)
     {
-        auto assetInfoFilePath = AZ::IO::Path{ sourceAssetPath };
-        assetInfoFilePath.Native() += ".assetinfo";
-        return CreateSceneManifest(sourceAssetPath, sourceAssetPath.Native() + ".assetinfo", collider, visual);
+        AZStd::unordered_set<AZ::IO::Path> assetsFilepaths;
+
+        // Look for material files
+        static const AZStd::array<AZ::SceneAPI::DataTypes::IMaterialData::TextureMapType, 9> allTextureTypes{
+            AZ::SceneAPI::DataTypes::IMaterialData::TextureMapType::Diffuse,
+            AZ::SceneAPI::DataTypes::IMaterialData::TextureMapType::Specular,
+            AZ::SceneAPI::DataTypes::IMaterialData::TextureMapType::Bump,
+            AZ::SceneAPI::DataTypes::IMaterialData::TextureMapType::Normal,
+            AZ::SceneAPI::DataTypes::IMaterialData::TextureMapType::Metallic,
+            AZ::SceneAPI::DataTypes::IMaterialData::TextureMapType::Roughness,
+            AZ::SceneAPI::DataTypes::IMaterialData::TextureMapType::AmbientOcclusion,
+            AZ::SceneAPI::DataTypes::IMaterialData::TextureMapType::Emissive,
+            AZ::SceneAPI::DataTypes::IMaterialData::TextureMapType::BaseColor
+        };
+        auto view =
+            AZ::SceneAPI::Containers::MakeDerivedFilterView<AZ::SceneAPI::DataTypes::IMaterialData>(scene->GetGraph().GetContentStorage());
+        for (const auto& material : view)
+        {
+            for (auto textureType : allTextureTypes)
+            {
+                if (const AZ::IO::Path filePath(material.GetTexture(textureType)); !filePath.empty())
+                {
+                    assetsFilepaths.emplace(filePath);
+                }
+            }
+        }
+        return assetsFilepaths;
```

Comment:
There is the assumption that materials are linked with meshes. So if a material is in the same package as mesh, it should work. I believe this assumption is valid, as the code handles only cases when materials are called from within a mesh.

Reply:
The assumption isn't _entirely_ correct. SDF files can have a <material> tag that references files directly, without being referenced by the mesh itself. However, this code looks like it will work well for the case that the mesh _does_ reference the files directly.

The point that I was trying to bring up though is that there are at least some material files (like the ambulance one) that aren't actually valid the way they're written, since they're dependent on something being able to resolve the "model://" prefixes of the file names. I'm not sure how these material files are authored or tested. But the problem we have with this with O3DE is that AssImpMaterialImporter::ResolveTexturePath() has its own heuristics on how to resolve a path, and those heuristics currently can't handle resolving "model://" or even for that matter finding ambulance.png because it lives in a sibling directory, not the same directory or a parent directory. I think we'll need to modify ResolveTexturePath() to allow for similar heuristics as the ones we have for the SDF files to be able to correctly resolve these.